test/DoctrineBatchUtilsTest/BatchProcessing/SelectBatchIteratorAggregateTest.php

```diff
@@ -0,0 +1,263 @@
+<?php
+
+namespace DoctrineBatchUtilsTest\BatchProcessing;
+
+use ArrayIterator;
+use Doctrine\Common\Persistence\Mapping\ClassMetadata;
+use Doctrine\ORM\AbstractQuery;
+use Doctrine\ORM\EntityManagerInterface;
+use DoctrineBatchUtils\BatchProcessing\SelectBatchIteratorAggregate;
+use PHPUnit\Framework\TestCase;
+use stdClass;
+
+/**
+ * @covers \DoctrineBatchUtils\BatchProcessing\SelectBatchIteratorAggregate
+ */
+final class SelectBatchIteratorAggregateTest extends TestCase
+{
+    /**
+     * @var AbstractQuery|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $query;
+
+    /**
+     * @var EntityManagerInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $entityManager;
+
+    /**
+     * @var ClassMetadata|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $metadata;
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function setUp()
+    {
+        $this->query         = $this->createMock(AbstractQuery::class);
+        $this->entityManager = $this->createMock(EntityManagerInterface::class);
+        $this->metadata      = $this->createMock(ClassMetadata::class);
+
+        $this->query->expects(self::any())->method('getEntityManager')->willReturn($this->entityManager);
```

Comment:
One thing to be added is `this->entityManager->expects(self::never())->method('flush');`